connectors/byoc.py

```diff
@@ -125,114 +133,144 @@ async def update_filtering_validation(
             retry_on_conflict=RETRY_ON_CONFLICT,
         )
 
-    def build_docs_query(self, native_service_types=None, connectors_ids=None):
-        if native_service_types is None:
-            native_service_types = []
-        if connectors_ids is None:
-            connectors_ids = []
-
-        if len(native_service_types) == 0 and len(connectors_ids) == 0:
+    async def supported_connectors(
+        self, native_service_types=None, connectors_ids=None
+    ):
+        query = supported_connectors_query(native_service_types, connectors_ids)
+        if query is None:
             return
 
-        native_connectors_query = {
-            "bool": {
-                "filter": [
-                    {"term": {"is_native": True}},
-                    {"terms": {"service_type": native_service_types}},
-                ]
-            }
-        }
-
-        custom_connectors_query = {
-            "bool": {
-                "filter": [
-                    {"term": {"is_native": False}},
-                    {"terms": {"_id": connectors_ids}},
-                ]
-            }
-        }
-        if len(native_service_types) > 0 and len(connectors_ids) > 0:
-            query = {
-                "bool": {"should": [native_connectors_query, custom_connectors_query]}
-            }
-        elif len(native_service_types) > 0:
-            query = native_connectors_query
-        else:
-            query = custom_connectors_query
-
-        return query
+        return self._get_all_docs(query=query)
 
     def _create_object(self, doc_source):
         return Connector(
             self,
             doc_source["_id"],
             doc_source["_source"],
-            bulk_options=self.bulk_options,
         )
 
 
-class SyncJob:
-    def __init__(self, elastic_index, connector_id, doc_source=None):
-        self.connector_id = connector_id
-        self.elastic_index = elastic_index
-        self.created_at = datetime.now(timezone.utc)
-        self.completed_at = None
-        self.job_id = None
-        self.status = None
-        self.client = elastic_index.client
+class ESDocument:
+    def __init__(self, elastic_index, doc_source=None):
+        self.index = elastic_index
         if doc_source is None:
             doc_source = dict()
+        self.id = doc_source.get("_id")
+        self._source = doc_source.get("_source", {})
+
+    def get(self, *keys, default=None):
+        value = self._source
+        for key in keys:
+            if not isinstance(value, dict):
+                return default
+            value = value.get(key)
```

Comment:
if you have several keys, `value` is overridden on the second round of this loop. is this intended?

Reply:
Overriding the variable `value` also means you won't have `self._source` in it.

Reply:
Yes, it's intended. 

> Overriding the variable `value` also means you won't have `self._source` in it.

Yes, we only need `value` pointing to `self._source` in the beginning, then we want to dig with `keys` provided, it's like the [dig method in Ruby Hash](https://ruby-doc.org/core-3.1.1/Hash.html#method-i-dig)

Reply:
ok I see 👍